### DISCUSSION TITLE
Adding remark42 to my blog

### DISCUSSION THREAD

**Discussion by criticalcake** [umputun/remark42#864] [Q&A] [answered]
Hi guys,

I'm new to hosting services and my own site.
It would be very kind, if you could help me out, to get remark42 working on my blog.

For my blog I subscribed to a webhosting offer, which works fine.
Since I cannot host docker containers with my subscription, I had to rent a vps. 
As I am new to vps, I spent several weeks to set this up properly.

This is my setup:
- Domain registration through my webhosting subscription
- In my DNS-settings *, @ and remark42 are pointing to my vps
- vps with ubuntu 20.04, nginx, ufw, fail2ban, docker-compose, certbot
- I followed several guides on digital ocean and elsewhere for the initial setup
- Also I tried to restrict access to my vps wherever possible and put effort in securing my environment
- With ufw I only allow access to OpenSSH, 443/tcp and a special port for remark42 (e.g. 12345)
- My blog works fine and is reachable through https
- My remark42-container is running fine as there are no errors in my docker logs

However the remark42 "quick installation test" fails, https://remark42.mydomain.com/web is not available.
Due to my lack of knowledge, I assume, there might be a wrong configuration with my nginx settings.
I didn't touch my nginx.conf file, but I added configuration-files in /etc/nginx/sites-available (which I attached):
mydomain.com
remark42.mydomain.com

[mydomain.com.txt](https://github.com/umputun/remark42/files/5860769/mydomain.com.txt)
[remark42.mydomain.com.txt](https://github.com/umputun/remark42/files/5860770/remark42.mydomain.com.txt)

I know they look different to the nginx configuration in the remark42 repository:
[https://github.com/umputun/remark42/blob/master/docs/latest/nginx-proxy.md](https://github.com/umputun/remark42/blob/master/docs/latest/nginx-proxy.md)

But I didn't want to screw up any of the certbot settings. The remark42-nginx-documentation states:
> note: proxy_pass points to internal DNS name remark42 and expected to run from the same compose. If nginx runs outside of compose the real IP (or docker's bridge IP) should be used

So maybe I use the wrong proxy_pass ip?
I tried all of the following values, but with no success:

proxy_pass              https://45.69.156.134/;              # example for my vps ip
proxy_pass              https://45.69.156.134:12345/;   # example ip and port
proxy_pass              https://mydomain.com:12345/;
proxy_pass              https://remark42.mydomain/;
proxy_pass              https://remark42.mydomain:12345/;
proxy_pass              https://172.18.0.2:12345/;        # ip and port of my remark42-container
proxy_pass              https://172.18.0.2/;

In my docker-compose.yml I've specified REMARK_PORT=12345
I cannot figure out, how to even get https://remark42.mydomain.com/web working.

Any ideas? Could someone please point me to the right direction?

Many thanks in advance!

**Comment by paskal** [comment]
Please post docker-compose logs output for your container to pastebin, it will help investigation greatly.

**Comment by criticalcake** [comment]
@paskal I didn't know about pastebin, great tool, thanks for sharing!

Here are the logs, configuration files or any other output, that might be helpful:

[docker-compose logs](https://pastebin.com/Yk5serjC)
[docker-compose.yml](https://pastebin.com/14FCK27w)
[nginx configuration for mydomain.com](https://pastebin.com/Th1DU8tU)
[nginx configuration for remark42.mydomain.com](https://pastebin.com/DVTDwwnA)
[ufw status verbose](https://pastebin.com/wUnseM1s)
[ss -tulpn](https://pastebin.com/esaRvpBH)

In the docker-compose logs, there is a warning that the cert file was not found (https server terminated).
I double checked the permisions, which seem to be fine and also the remark42-container  runs with APP_UID=0 (=root).
So I don`t know how to fix this warning. But then again, it's just a warning.
I assume, that at least the default "/web" demo page should work under that warning, right?

But keep in mind, that I try to get the remark42 container working as a non-root user.
Please let me know, if you need any other data and I will provide it through pastebin.

Any help is very much appreciated!

**Comment by criticalcake** [comment]
I solved the ssl warning by changing the SSL_TYPE to 'auto' in my docker-compose.yml:

[docker-compose logs with ssl_type=auto](https://pastebin.com/vN1FQmTT)

Unfortunately the remark42-demo page still doesn't show up.
I assume, it must be available under https://remark42.mydomain.com/web
But I get _400 Bad Request - Request Header Or Cookie Too Large_

**Comment by paskal** [comment]
Problem in the first docker-compose log: `[WARN]  {api/rest.go:122 api.(*Rest).Run} https server terminated, open /etc/letsencrypt/live/mydomain.com/fullchain.pem: no such file or directory`, after which remark42 is terminated.

In the second log, I saw no access entries so requests don't seem to reach it. After you changed the settings I think the problem is that your Nginx tries to connect to itself instead of connecting to the container, as your container [is not exposing any ports](https://docs.docker.com/compose/compose-file/compose-file-v2/#ports) to the host machine.

Remove following lines from your [docker-compose.yml](https://pastebin.com/14FCK27w) to disable HTTPS termination inside remark42 container, as we want your nginx to handle the HTTPS part and talk to remark backend by HTTP.
```yaml
            - SSL_TYPE=static
            - SSL_PORT=443
            - SSL_CERT=/etc/letsencrypt/live/mydomain.com/fullchain.pem
            - SSL_KEY=/etc/letsencrypt/live/mydomain.com/privkey.pem
```

Then expose port 8080 of remark container to the host but only locally, to do that add the following lines to the same container configuration:
```
        ports:
           - "127.0.0.1:8080:8080"
```

The last step is fixing [nginx configuration](https://pastebin.com/DVTDwwnA) `proxy_pass` to point to remark42 container exposed locally on port 8080:
```
proxy_pass              http://localhost:8080/;
```

After that please try again and if it doesn't work, please post the docker-compose logs of the container and nginx logs after few attempts to reach the comments.

**Comment by criticalcake** [comment, marked as answer]
@paskal Thank you very much for your help! 🙏 

I followed your suggestions and modified my docker-compose.yml and the nginx configuration for remark42.mydomain.com.
Unfortunately it still doesn't work, although I cleared the browser cache and also changed the browser. 😢
I removed REMARK_PORT=12345 from the docker-compose.yml, but still no success.

[docker-compose logs 2](https://pastebin.com/exV0FZVv)
[nginx remark42.log](https://pastebin.com/iPeGRjcn)
[nginx error.log](https://pastebin.com/mVxMAgnZ)
[nginx access.log](https://pastebin.com/LS1M36hH)

**Reply by paskal** [reply]
What I see in the logs:

```
2021/01/24 16:45:46 [error] 50723#50723: *1 connect() failed (111: Connection refused) while connecting to upstream, client: 84.58.178.141, server: remark42.mydomain.com, request: "GET /web HTTP/1.1", upstream: "http://[::1]:8080/web", host: "remark42.mydomain.com"
2021/01/24 16:45:46 [error] 50723#50723: *1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: *.*.*.*, server: remark42.mydomain.com, request: "GET /web HTTP/1.1", upstream: "http://127.0.0.1:8080/web", host: "remark42.mydomain.com"
2021/01/24 16:45:46 [error] 50723#50723: *1 no live upstreams while connecting to upstream, client: *.*.*.*, server: remark42.mydomain.com, request: "GET /favicon.ico HTTP/1.1", upstream: "http://localhost/favicon.ico", host: "remark42.mydomain.com", referrer: "https://remark42.mydomain.com/web"
```

Can you please try to follow [this advice](https://serverfault.com/a/796912/118600) and add `resolver 8.8.8.8 ipv6=off;` to remark42 Nginx `server` block?

**Reply by criticalcake** [reply]
Thanks for your quick reply! Your suggestion that these errors might have something to do with ipv6 is correct.
I did some research on this and found another comment on this matter:

> writing something like
> 
> proxy_pass http://127.0.0.1:8080/files/;
> 
> with "127.0.0.1" IPv4 address explicitly used instead of
> "localhost" should be enough.

So I changed the value of `proxy_pass` in my remark42 Nginx `server` block
from `http://localhost:8080/;`
to `http://127.0.0.1:8080/;`
Now all of the error messages in the nginx logs are gone! 👍

When I open https://remark42.mydomain.com/web/ I get a 404, which looks promising to me.
I assume, that remark42 is now reachable from an external source, but Nginx is not looking in the correct path?

**Reply by paskal** [reply]
Please check docker-compose logs, if you see hits then that's remark42 giving you 404s.

**Reply by criticalcake** [reply]
Sadly, the 404 is not from the remark42 container. There are no recently added lines in the docker-compose logs. 😕

**Reply by paskal** [reply]
Please show Nginx access and error log for remark42 site then, please.

**Reply by criticalcake** [reply]
Hello @paskal,

tried to fix the problem for myself, but no success :-/
No new lines in docker-compose logs.
For Nginx there are no recent lines in the error log.

Although I have to Nginx access logs I can share, they don't contain anything strange:
[Nginx access.log](https://pastebin.com/6citVDvE)
[Nginx remark42.access.log](https://pastebin.com/0aiKYeJH)  => This is created within the Nginx remark42 server block

I checked the IP for the remark42 container:
`docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' remark42`
which is `172.18.0.2`
I changed the value from `proxy_pass` to this IP and checked the results, but I get 502 Bad Gateway,
so I suppose `127.0.0.1:8080/` is the correct value for `proxy_pass`.

**Reply by paskal** [reply]
Please add log format `apm` from [there](https://github.com/paskal/bitrix.infra/blob/master/config/nginx/nginx.conf#L24-L38) and enable it with changing `access_log  /var/log/nginx/remark42.log;` to `access_log  /var/log/nginx/remark42.log apm;`, and then posting the remark42 single log line which is currently listed as `/web/ HTTP/1.1" 404`: with new log format we'll see the upstream it's hitting.

**Reply by criticalcake** [reply]
Nginx exits with 1/failure after activating apm. Debugging my nginx configuration will take some time...

**Reply by paskal** [reply]
Maybe the version is too old for some option, it should tell you precisely what is wrong with that I assume. Please note that log_format [should be in the http block](https://nginx.org/en/docs/http/ngx_http_log_module.html#log_format), not in `server`.

**Reply by criticalcake** [reply]
I placed it in the http block of nginx.conf, but will check the docs step by step.
The journalctl -xe states simply "invalid log level 'apm'". That's it.
```
Jan 25 22:26:10 myvps sudo[4888]:     myuser : TTY=pts/0 ; PWD=/mnt/vg1/logs/nginx/mydomain.com ; USER=root ; COMMAND=/usr/bin/systemctl start nginx
Jan 25 22:26:10 myvps sudo[4888]: pam_unix(sudo:session): session opened for user root by myuser(uid=0)
Jan 25 22:26:10 myvps systemd[1]: Starting A high performance web server and a reverse proxy server...
-- Subject: A start job for unit nginx.service has begun execution
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- A start job for unit nginx.service has begun execution.
--
-- The job identifier is 1391.
Jan 25 22:26:10 myvps nginx[4891]: nginx: [emerg] invalid log level "apm" in /etc/nginx/sites-enabled/remark42.mydomain.com:16
Jan 25 22:26:10 myvps nginx[4891]: nginx: configuration file /etc/nginx/nginx.conf test failed
Jan 25 22:26:10 myvps systemd[1]: nginx.service: Control process exited, code=exited, status=1/FAILURE
-- Subject: Unit process exited
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- An ExecStartPre= process belonging to unit nginx.service has exited.
--
-- The process' exit code is 'exited' and its exit status is 1.
Jan 25 22:26:10 myvps systemd[1]: nginx.service: Failed with result 'exit-code'.
-- Subject: Unit failed
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- The unit nginx.service has entered the 'failed' state with result 'exit-code'.
Jan 25 22:26:10 myvps systemd[1]: Failed to start A high performance web server and a reverse proxy server.
-- Subject: A start job for unit nginx.service has failed
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- A start job for unit nginx.service has finished with a failure.
--
-- The job identifier is 1391 and the job result is failed.
Jan 25 22:26:10 myvps sudo[4888]: pam_unix(sudo:session): session closed for user root
```

**Reply by criticalcake** [reply]
Now Nginx restarts and is running without any errors, but I cannot see any changes in log_format. Will check this tomorrow with a fresh mind... 

Thank you very much for your support!

**Reply by criticalcake** [reply]
Hi @paskal, 
I've got only a few minutes of time, but I wanted you to know, that apm is working fine now:
[Nginx access.log with apm](https://pastebin.com/35WApWx3)
[Nginx remark42.access.log with apm](https://pastebin.com/UbaVYcCb)

If I understand the apm access logs correctly, I would assume that there is no upstream communication for remark42.
I still do not have any new lines neither in the error logs nor in the docker-compose logs.

Maybe `ufw` denies the upstream?
```
Status: active
Logging: on (low)
Default: deny (incoming), allow (outgoing), deny (routed)
New profiles: skip

To                         Action      From
--                         ------      ----
12344/tcp                  ALLOW IN    Anywhere
22/tcp (OpenSSH)           ALLOW IN    Anywhere
443/tcp                    ALLOW IN    Anywhere
12345/tcp                  ALLOW IN    Anywhere
8080                       ALLOW IN    Anywhere
12344/tcp (v6)             ALLOW IN    Anywhere (v6)
22/tcp (OpenSSH (v6))      ALLOW IN    Anywhere (v6)
443/tcp (v6)               ALLOW IN    Anywhere (v6)
12345/tcp (v6)             ALLOW IN    Anywhere (v6)
8080 (v6)                  ALLOW IN    Anywhere (v6)
```

**Reply by criticalcake** [reply]
Hi @paskal,

> Please add log format apm from there and then posting the remark42 single log line which is currently listed as /web/ HTTP/1.1" 404: with new log format we'll see the upstream it's hitting.

Here are the 404 lines:
```
"27/Jan/2021:18:18:16 +0100" client=84.58.178.141 method=GET request="GET /web/embed.js HTTP/1.1" request_length=326 status=404 bytes_sent=561 body_bytes_sent=19 referer=https://mydomain.com/blog/pinnwand/ user_agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:84.0) Gecko/20100101 Firefox/84.0" upstream_addr=127.0.0.1:8080 upstream_status=404 request_time=0.002 upstream_response_time=0.000 upstream_connect_time=0.000 upstream_header_time=0.000
"27/Jan/2021:18:18:28 +0100" client=84.58.178.141 method=GET request="GET /web/ HTTP/1.1" request_length=370 status=404 bytes_sent=561 body_bytes_sent=19 referer=- user_agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:84.0) Gecko/20100101 Firefox/84.0" upstream_addr=127.0.0.1:8080 upstream_status=404 request_time=0.001 upstream_response_time=0.000 upstream_connect_time=0.000 upstream_header_time=0.000
"27/Jan/2021:18:18:28 +0100" client=84.58.178.141 method=GET request="GET /favicon.ico HTTP/1.1" request_length=335 status=404 bytes_sent=329 body_bytes_sent=19 referer=https://remark42.mydomain.com/web/ user_agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:84.0) Gecko/20100101 Firefox/84.0" upstream_addr=127.0.0.1:8080 upstream_status=404 request_time=0.002 upstream_response_time=0.000 upstream_connect_time=0.000 upstream_header_time=0.000
```

I would assume that the upstream is not hitting. Correct?
Any ideas how I could fix this?

Many thanks in advance!

**Reply by paskal** [reply]
According to that nginx hit `127.0.0.1:8080` and it responded with 404. Could you please check output of `curl -I http://127.0.0.1:8080/web/` on that host? I am working under the assumption that nginx is running on the host and not inside the container, and should be able to see the container port 8080 only when it's exposed to host.

Also please check what is listening on port 8080 according to [that](https://unix.stackexchange.com/a/106562/56821) answer, we might see something surprising there.

**Reply by criticalcake** [reply]
Thank you so much for your kind help. Yes, Nginx is running on the host, not inside the container.

```
curl -I http://127.0.0.1:8080/web/

HTTP/1.1 405 Method Not Allowed
App-Name: remark42
App-Version: v1.6.0-71dcb13-20200426T13:40:13
Author: umputun
Vary: Origin
Date: Wed, 27 Jan 2021 19:38:13 GMT

sudo ss -lptn 'sport = :8080'
State                                 Recv-Q                                Send-Q                                                                Local Address:Port                                                                 Peer Address:Port                                Process
LISTEN                                0                                     4096                                                                      127.0.0.1:8080                                                                      0.0.0.0:*                                    users:(("docker-proxy",pid=1004,fd=4))

sudo lsof -n -i :8080 | grep LISTEN
docker-pr 1004 root    4u  IPv4  26779      0t0  TCP 127.0.0.1:http-alt (LISTEN)
```

The output seems correct to me, except for `HTTP/1.1 405 Method Not Allowed`
Maybe the problem relates to http/https?

**Reply by umputun** [reply]
nah, this is a HEAD request we don't support. But this hit looks fine to me. If you do `curl -i http://127.0.0.1:8080/web/` you should see HTML response, and if you do `curl -i http://127.0.0.1:8080/ping` you should get pong with headers. This means your remark42 runs fine and the problem is on proxy level (nginx)

**Reply by criticalcake** [reply]
```
curl -i http://127.0.0.1:8080/web/

HTTP/1.1 404 Not Found
App-Name: remark42
App-Version: v1.6.0-71dcb13-20200426T13:40:13
Author: umputun
Cache-Control: max-age=3600
Content-Type: text/plain; charset=utf-8
Etag: "fa2a847f85dfb08375c5f47fe63844a665f39e89"
Vary: Origin
X-Content-Type-Options: nosniff
X-Rate-Limit-Duration: 1
X-Rate-Limit-Limit: 20.00
X-Rate-Limit-Request-Forwarded-For:
X-Rate-Limit-Request-Remote-Addr: 172.18.0.1:56990
Date: Wed, 27 Jan 2021 20:02:42 GMT
Content-Length: 19

404 page not found
```

```
curl -i http://127.0.0.1:8080/ping

HTTP/1.1 200 OK
App-Name: remark42
App-Version: v1.6.0-71dcb13-20200426T13:40:13
Author: umputun
Content-Type: text/plain
Date: Wed, 27 Jan 2021 20:03:29 GMT
Content-Length: 4
```

So the container works fine and it`s a problem with Nginx?

**Reply by umputun** [reply]
Well, remark42 is running on :8080 and responds to API request (ping) but somehow web request routed to the wrong place. I suspect smth with your `REMARK_URL` param, but if you add the current compose and nginx part for remerk42 proxy - it may help to diagnose the issue

**Reply by criticalcake** [reply]
These are my configuration files:
[docker-compose.yml](https://pastebin.com/xTc9Rqhf)
[nginx configuration for remark42](https://pastebin.com/74rUw5Ta)

Thank you very much for your support!

**Reply by umputun** [reply]
I don't see anything obviously wrong in here. One unusual thing is SITE value, but as long as it matches your web side it should be ok. @akellbl4 maybe you see smth here I missed? For whatever reason routing to web assets seem to be broken.

**Reply by criticalcake** [reply]
I added debug mode for the nginx error logs. When I open one of my pages, where remark42 should show up, I get the following logs:
[nginx logs in debugging mode](https://pastebin.com/5gqRPcjE)

According to lines 120 and following:
```
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy status 404 "404 Not Found"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "App-Name: remark42"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "App-Version: v1.6.0-71dcb13-20200426T13:40:13"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "Author: umputun"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "Cache-Control: max-age=3600"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "Content-Type: text/plain; charset=utf-8"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "Etag: "7505f275aa8b7790c231916308b44535fee28d07""
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "Vary: Origin"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "X-Content-Type-Options: nosniff"
2021/01/30 22:36:08 [debug] 7364#7364: *16 posix_memalign: 0000562A5DDFE9C0:4096 @16
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "X-Rate-Limit-Duration: 1"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "X-Rate-Limit-Limit: 20.00"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "X-Rate-Limit-Request-Forwarded-For: 84.58.178.141"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "X-Rate-Limit-Request-Remote-Addr: 84.58.178.141"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "Date: Sat, 30 Jan 2021 21:36:08 GMT"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header: "Content-Length: 19"
2021/01/30 22:36:08 [debug] 7364#7364: *16 http proxy header done
2021/01/30 22:36:08 [debug] 7364#7364: *16 xslt filter header
2021/01/30 22:36:08 [debug] 7364#7364: *16 HTTP/1.1 404 Not Found
```

So although I just see my web page (and neither remark42 nor 404), there must be some sort of communication to remark42.

**Reply by akellbl4** [reply]
@criticalcake hey! let's check the frontend filed in the docker container.
try to do `docker exec %container_id ls -la web` (replace `%container_id` with your container id from `docker ps`)

**Reply by criticalcake** [reply]
@akellbl4 thanks for your reply!

I have good news, finaly:
I removed the `working_dir` parameter from my docker-compose.yml, now I have access to https://remark42.mydomain.com/web !

```
remark42@myvps:/opt/remark42$ docker exec 2he45b46bz71 ls -la web
total 1492
drwxr-xr-x    1 root     app           4096 Jan 31 13:08 .
drwxr-xr-x    1 root     app           4096 Jan 31 13:08 ..
-rw-r--r--    1 root     app           2166 Apr 26  2020 comments.html
-rw-r--r--    1 root     app           1479 Apr 26  2020 counter.html
-rw-r--r--    1 root     app           2324 Jan 31 13:08 counter.js
-rw-r--r--    1 root     app          10024 Jan 31 13:08 de.js
-rw-r--r--    1 root     app           1586 Apr 26  2020 deleteme.html
-rw-r--r--    1 root     app         114048 Jan 31 13:08 deleteme.js
-rw-r--r--    1 root     app           7461 Jan 31 13:08 embed.js
-rw-r--r--    1 root     app           9130 Jan 31 13:08 es.js
-rw-r--r--    1 root     app           8902 Jan 31 13:08 fi.js
drwxr-xr-x    1 root     app           4096 Apr 26  2020 files
-rw-r--r--    1 root     app           4133 Apr 26  2020 iframe.html
-rw-r--r--    1 root     app           3906 Apr 26  2020 index.html
-rw-r--r--    1 root     app           7086 Jan 31 13:08 intersection-observer.js
-rw-r--r--    1 root     app          25927 Apr 26  2020 last-comments.css
-rw-r--r--    1 root     app           1064 Apr 26  2020 last-comments.html
-rw-r--r--    1 root     app         155469 Jan 31 13:08 last-comments.js
-rw-r--r--    1 root     app           9792 Apr 26  2020 markdown-help.html
-rw-r--r--    1 root     app          44382 Apr 26  2020 remark.css
-rw-r--r--    1 root     app         323125 Jan 31 13:08 remark.js
-rw-r--r--    1 root     app         496110 Apr 26  2020 report.html
-rw-r--r--    1 root     app          12078 Jan 31 13:08 ru.js
-rw-r--r--    1 root     app         159975 Jan 31 13:08 vendors~core-js.js
-rw-r--r--    1 root     app          42605 Jan 31 13:08 vendors~node-emoji.js
-rw-r--r--    1 root     app           7868 Jan 31 13:08 whatwg-fetch.js
-rw-r--r--    1 root     app           8028 Jan 31 13:08 zh.js
```

So finally this is working for /web.
However I cannot add a comment at /web. The docker-compose logs tell me, that `site "remark" not found`.
I assume "remark" is the default site id. However I have configured `site=mydomain` in my docker-compose.yml

```
remark42    | 2021/01/31 13:09:13.181 [WARN]  {rest/httperrors.go:85 rest.SendErrorJSON} can't get last comments - site "remark" not found - 500 (0) - /api/v1/last/50?site=remark - [rest/api/rest_public.go:234 api.(*public).lastCommentsCtrl]
remark42    | 2021/01/31 13:09:13.182 [INFO]  {logger/logger.go:122 logger.(*Middleware).Handler.func1.1} GET - /api/v1/last/50?site=remark - 36a8b0691a15 - 500 (83) - 409.051µs
remark42    | 2021/01/31 13:09:13.199 [INFO]  {logger/logger.go:122 logger.(*Middleware).Handler.func1.1} GET - /api/v1/config?site=remark - 36a8b0691a15 - 200 (381) - 171.988µs
remark42    | 2021/01/31 13:09:13.242 [DEBUG] {middleware/auth.go:69 middleware.(*Authenticator).auth.func1} auth failed, can't get token: token cookie was not presented: http: named cookie not present
remark42    | 2021/01/31 13:09:13.244 [DEBUG] {api/rest_public.go:74 api.(*public).findCommentsCtrl} get comments for {SiteID:remark URL:https://remark42.com/demo/}, sort -active, format tree, since 0001-01-01 00:00:00 +0000 UTC
remark42    | 2021/01/31 13:09:13.245 [INFO]  {logger/logger.go:122 logger.(*Middleware).Handler.func1.1} GET - /api/v1/find?site=remark&url=https://remark42.com/demo/&sort=-active&format=tree - 36a8b0691a15 - 200 (115) - 708.795µs
remark42    | 2021/01/31 13:09:22.960 [INFO]  {logger/logger.go:122 logger.(*Middleware).Handler.func1.1} POST - /api/v1/counts?site=remark - 505b5910118f - 200 (3) - 88.169µs - ["https://remark42.com/demo/"]
```

This should be working even if it's the demo page. Correct?

**Reply by akellbl4** [reply]
Good to hear.

Yep it should work, I tried to fix it but as I see it isn't successful. I pay my attention it and will try to fix it in one of the next releases.

**Reply by criticalcake** [reply]
I changed my site id to `remark`, now the demo site is working fine. 👍 

Unfortunately remark42 is still not displayed on my blog.
The Docker and Nginx logs look just fine. No Errors, no warnings.
Maybe I use the wrong directories?

```
/opt/remark42/docker-compose.yml
/mnt/vg1/mydomain.com/html/index.html (=> my blog)
```

server block for my website
```
root /mnt/vg1/mydomain.com/html;

location / {
   try_files $uri $uri/ =404;
}
```

server block for remark42
```
        location / {
                proxy_redirect          off;
                proxy_set_header        X-Real-IP $remote_addr;
                proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header        Host $http_host;
                proxy_pass              http://127.0.0.1:8080/;
                proxy_read_timeout      1m;
                proxy_connect_timeout   1m;
        }
```
Do I have to use `root /mnt/vg1/mydomain.com/html;` here too?

docker-compose.yml
```
        ports:
            - "127.0.0.1:8080:8080"

        environment:
            - REMARK_URL=https://remark42.mydomain.com
            - SECRET=**************
            - SITE=remark
            - STORE_TYPE=bolt
            - STORE_BOLT_PATH=/mnt/vg1/remark42/var/db
        ...
        volumes:
                - /mnt/vg1/remark42:/srv/var
```

Anything suspicious here?
Do I use the paths/directories like they are intended to?

**Reply by criticalcake** [reply]
Guys, remark42 is working fine now on my blog! 🎉 😃 

After the demo page was working, I checked my frontend and recognized the following error:
```
Uncaught TypeError: window.remark_config is undefined
    a https://remark42.mydomain.com/web/embed.js:1
    126 https://remark42.mydomain.com/web/embed.js:1
    n https://remark42.mydomain.com/web/embed.js:1
    <anonymous> https://remark42.mydomain.com/web/embed.js:1
    <anonymous> https://remark42.mydomain.com/web/embed.js:1
```

In the script part of my html I used `var remark42_config = ...`
So I had to change this to `var remark_config = ...` (without 42). Now it works perfect!

Thank you very much for your kind help and professional support! 👏 👍
I will recommend your engine and write a blog post about it.

**Comment by criticalcake** [comment]
wait: Do I need to have the ngx_http_log_module installed?

**Reply by paskal** [reply]
It's impossible not to have it installed, it's part of the base nginx installation.
